gradle/native-build.gradle

```diff
@@ -0,0 +1,136 @@
+if (System.getProperty('os.name').toLowerCase().contains("mac")) {
+    new ByteArrayOutputStream().withStream { os ->
+        exec {
+            args '-version', '-sdk', 'iphoneos', 'SDKVersion'
+            executable 'xcodebuild'
+            standardOutput = os
+        }
+        ext.IOS_VERSION = os.toString().trim()
+    }
+} else {
+    ext.IOS_VERSION = ""
+}
+
+def sdkPath(String platform) {
+    return "/Applications/Xcode.app/Contents/Developer/Platforms/${platform}.platform/Developer/SDKs/${platform}${IOS_VERSION}.sdk";
+}
+
+ext.nativeBuild = { buildDir, projectDir, name, os ->
+
+    println("native build for $name  started")
+    File shareDir = file("$projectDir/src/main/native/share")
+    if (!shareDir.exists()) {
+        // TODO
+        //println("No share lib dir found for $name")
+        //return
+    }
+    def sharedSources = shareDir.listFiles()
+
+    def osSources = []
+    File osDir = file("$projectDir/src/main/native/$os")
+    if (osDir.exists()) {
+        osSources = osDir.listFiles()
+    } else {
+        println("No native lib dir found for $name at $osDir")
+        return
+    }
+
+    def JAVAHOME = System.getenv("JAVA_HOME")
+    def includeFlags = "-I$JAVAHOME/include"
+    def osIncludeFlags = ""
+
+    if (os == "ios") {
+        osIncludeFlags = "-I$JAVAHOME/include/darwin"
+
+        def linkerOutputs = []
+        def lipoOutput = "$buildDir/native/ios/lib${name}.a"
+
+        def buildSystems = ["iPhoneOS+arm64", "iPhoneSimulator+x86_64"]
+        buildSystems.each { buildSystem ->
+
+            def (platform, arch) = buildSystem.tokenize("+")
+            if (!file(sdkPath(platform)).exists()) {
+                println "Skipping native-build on iOS: path for ${platform} doesn't exist"
+                return
+            }
+            
+            osSources = "$projectDir/src/main/native/$os/${name}.m"
+            
+            def compileOutput = "$buildDir/native/ios/$arch"
+            new File(compileOutput).mkdirs()
+            def compileOutputs = ["$buildDir/native/ios/$arch/${name}.o"]
+            def linkerOutput = "$buildDir/native/ios/$arch/lib${name}.a"
+            
+            def clangArgs = [
+                "-xobjective-c",
+                "-mios-version-min=11.0",
+                includeFlags,
+                osIncludeFlags,
+                "-c",
+                "-fPIC",
+                "-arch", arch,
+                "-isysroot",
+                sdkPath(platform),
+                osSources].flatten()
+
+            exec {
+                executable "clang"
+                args clangArgs
+                workingDir compileOutput
+            }
+
+            def linkerArgs = [
+                    "-static",
+                    "-framework", "Foundation",
```

Comment:
Can we make this list plugin-specific? E.g. UserNotifications won't be needed in all plugins

Reply:
Actually the frameworks used will be defined by the plugin, won't they?

We had the same issue with Charm Down and jfxmobile plugin, where we defined some plugins in Down in order to build the services, and the latter defined the frameworks that were ultimately used and even allowed an extension of these frameworks.